# deepsignal2 rerun arab part2 254.2 ============
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --result_file athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.freq.p0.8.tsv --prob_cf 0.8  > athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.freq.p0.8.log &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --result_file athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.freq.p0.8.tsv --prob_cf 0.8  > athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.freq.p0.8.log &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.freq.p0.8.tsv --prob_cf 0.8  > athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.freq.p0.8.log &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --result_file athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.freq.p0.8.tsv --prob_cf 0.8  > athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.freq.p0.8.log &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.freq.p0.8.tsv --prob_cf 0.8  > athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.freq.p0.8.log &

python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CG.deepsignal2_comb.p0.8.diffcov_vs_bsrep123.tsv --contig_prefix NC_003 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CG.deepsignal2_comb.p0.8.diffcov_vs_bsrep123.log 2>&1 &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CHG.deepsignal2_comb.p0.8.diffcov_vs_bsrep123.tsv --contig_prefix NC_003 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CHG.deepsignal2_comb.p0.8.diffcov_vs_bsrep123.log 2>&1 &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --nanofile athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CHH.deepsignal2_comb.p0.8.diffcov_vs_bsrep123.tsv --contig_prefix NC_003 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CHH.deepsignal2_comb.p0.8.diffcov_vs_bsrep123.log 2>&1 &

python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --covcf 5 --contig_prefix NC_003 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.vs_bsrep123.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.vs_bsrep123.rmet.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --covcf 5 --contig_prefix NC_003 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bsrep123.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bsrep123.rmet.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --covcf 5 --contig_prefix NC_003 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bsrep123.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bsrep123.rmet.log 2>&1 &

# megalodon rerun arab part2 254.2 ============
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.arab.pass.part2_guppy.10x_1.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_2.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_3.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_4.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CG.megalodon_bscov15.p0.8.diffcov_vs_bsrep123.tsv --contig_prefix NC_003 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CG.megalodon_bscov15.p0.8.diffcov_vs_bsrep123.log 2>&1 &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.arab.pass.part2_guppy.10x_1.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_2.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_3.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_4.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CHG.megalodon_bscov15.p0.8.diffcov_vs_bsrep123.tsv --contig_prefix NC_003 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CHG.megalodon_bscov15.p0.8.diffcov_vs_bsrep123.log 2>&1 &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.arab.pass.part2_guppy.10x_1.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_2.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_3.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_4.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.arab.pass.part2_guppy.10x_5.CHH_retrain_comb/modified_bases.5mC.bed --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CHH.megalodon_bscov15.diffcov_vs_bsrep123.tsv --contig_prefix NC_003 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass2.CHH.megalodon_bscov15.diffcov_vs_bsrep123.log 2>&1 &



python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.arab.pass.part2_guppy.10x_1.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_2.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_3.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_4.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile megalodon_results.arab.pass.part2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed 
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.arab.pass.part2_guppy.10x_1.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_2.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_3.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_4.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile megalodon_results.arab.pass.part2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.arab.pass.part2_guppy.10x_1.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_2.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_3.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_4.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.arab.pass.part2_guppy.10x_5.CHH_retrain_comb/modified_bases.5mC.bed --wfile megalodon_results.arab.pass.part2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed &

python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.arab.pass.part2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --covcf 5 --contig_prefix NC_003 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.arab.pass.part2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.vs_bsrep123.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.arab.pass.part2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.vs_bsrep123.rmet.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.arab.pass.part2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --covcf 5 --contig_prefix NC_003 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.arab.pass.part2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.vs_bsrep123.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.arab.pass.part2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.vs_bsrep123.rmet.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.arab.pass.part2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --covcf 5 --contig_prefix NC_003 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.arab.pass.part2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.vs_bsrep123.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.arab.pass.part2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.vs_bsrep123.rmet.log 2>&1 &





# deepsignal2 rerun rice1-1 ==============
# CG
CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.1 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 20 --nproc_gpu 4 > shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.log 2>&1

CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.2 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.log 2>&1

CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.log 2>&1

CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.4 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.log 2>&1

CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.5 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 20 --nproc_gpu 4 > shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.log 2>&1


# CHG
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.1 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 20 --nproc_gpu 4 > shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.log 2>&1

# test rice1-1.10x.2 =
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.2 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.log 2>&1

# test rice1-1.10x.3 =
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.log 2>&1

# test rice1-1.10x.4 =
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.4 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.log 2>&1

# test rice1-1.10x.5 =
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.5 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.log 2>&1



# CHH
# test rice1-1.10x =
CUDA_VISIBLE_DEVICES=3 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.1 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.log 2>&1

# test rice1-1.10x.2 =
CUDA_VISIBLE_DEVICES=3 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.2 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 25 --nproc_gpu 5 > shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.log 2>&1

# test rice1-1.10x.3 =
CUDA_VISIBLE_DEVICES=3 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.log 2>&1

# test rice1-1.10x.4 =
CUDA_VISIBLE_DEVICES=3 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.4 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.log 2>&1

# test rice1-1.10x.5 =
CUDA_VISIBLE_DEVICES=3 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.5 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.log 2>&1


# 254.3
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.tsv --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.tsv --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.tsv --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.tsv --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.tsv --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.tsv --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.tsv --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.tsv --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.tsv --input_path shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.tsv --result_file shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --prob_cf 0.8 > shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CG.deepsignal2_comb.p0.8.diffcov_vs_bs1-2.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CG.deepsignal2_comb.p0.8.diffcov_vs_bs1-2.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.vs_bs1-2.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.balance.50x_12345.p0.8.freq.vs_bs1-2.rmet.log 2>&1 &



python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --input_path shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --prob_cf 0.8 > shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CHG.deepsignal2_comb.p0.8.diffcov_vs_bs1-2.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CHG.deepsignal2_comb.p0.8.diffcov_vs_bs1-2.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bs1-2.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bs1-2.rmet.log 2>&1 &




python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --input_path shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --prob_cf 0.8 > shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --nanofile shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CHH.deepsignal2_comb.p0.8.diffcov_vs_bs1-2.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CHH.deepsignal2_comb.p0.8.diffcov_vs_bs1-2.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bs1-2.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bs1-2.rmet.log 2>&1 &





# megalodon rerun rice1-1 part2 254.3 ============
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.rice.pass2_guppy.10x_1.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_2.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_3.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_4.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CG.megalodon_bscov15.diffcov_vs_bs1-2.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CG.megalodon_bscov15.diffcov_vs_bs1-2.log 2>&1 &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.rice.pass2_guppy.10x_1.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_2.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_3.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_4.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CHG.megalodon_bscov15.diffcov_vs_bs1-2.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CHG.megalodon_bscov15.diffcov_vs_bs1-2.log 2>&1 &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.rice.pass2_guppy.10x_1.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_2.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_3.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_4.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.rice.pass2_guppy.10x_5.CHH_retrain_comb/modified_bases.5mC.bed --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CHH.megalodon_bscov15.diffcov_vs_bs1-2.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao1-1.guppy.pass2.CHH.megalodon_bscov15.diffcov_vs_bs1-2.log 2>&1 &

python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.rice.pass2_guppy.10x_1.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_2.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_3.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_4.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile megalodon_results.rice.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.rice.pass2_guppy.10x_1.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_2.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_3.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_4.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile megalodon_results.rice.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.rice.pass2_guppy.10x_1.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_2.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_3.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_4.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.rice.pass2_guppy.10x_5.CHH_retrain_comb/modified_bases.5mC.bed --wfile megalodon_results.rice.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed &

python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.rice.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice1-1.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.vs_bs1-2.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice1-1.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.vs_bs1-2.rmet.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.rice.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice1-1.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.vs_bs1-2.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice1-1.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.vs_bs1-2.rmet.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.rice.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice1-1.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.vs_bs1-2.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice1-1.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.vs_bs1-2.rmet.log 2>&1 &





# deepsignal2 run rice2-1 ==============
# CG
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.1 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.log 2>&1
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.2 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.log 2>&1
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.log 2>&1

CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.4 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.log 2>&1
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.5 --model_path model.comb/model.dp2.CG.arabnrice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch6.ckpt --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.log 2>&1

# CHG
CUDA_VISIBLE_DEVICES=1 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.1 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.log 2>&1
CUDA_VISIBLE_DEVICES=1 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.2 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.log 2>&1

CUDA_VISIBLE_DEVICES=1 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.log 2>&1

CUDA_VISIBLE_DEVICES=1 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.4 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.log 2>&1
CUDA_VISIBLE_DEVICES=1 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.5 --model_path model.comb/model.dp2.CHG.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.log 2>&1

# CHH
CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.1 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.log 2>&1
CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.2 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.log 2>&1

CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 30 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.log 2>&1

CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.4 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.log 2>&1
CUDA_VISIBLE_DEVICES=2 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.5 --model_path model.comb/model.dp2.CHH.arabnrice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.log 2>&1


# 254.3
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.tsv --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.tsv --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.tsv --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.tsv --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.tsv --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.tsv --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.tsv --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.tsv --input_path shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.tsv --result_file shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --prob_cf 0.8 > shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_2.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_4.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.deepsignal2_comb.p0.8.diffcov_vs_bs2-1.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.deepsignal2_comb.p0.8.diffcov_vs_bs2-2.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.vs_bs2-1.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.vs_bs2-1.rmet.log 2>&1 &

python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --input_path shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --prob_cf 0.8 > shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHG.deepsignal2_comb.p0.8.diffcov_vs_bs2-1.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHG.deepsignal2_comb.p0.8.diffcov_vs_bs2-1.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bs2-1.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bs2-1.rmet.log 2>&1 &

python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.tsv --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.tsv --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.tsv --input_path shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.tsv --result_file shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --prob_cf 0.8 > shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_1.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_2.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_4.p0.8.freq.tsv --nanofile shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.10x_5.p0.8.freq.tsv --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHH.deepsignal2_comb.p0.8.diffcov_vs_bs2-1.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHH.deepsignal2_comb.p0.8.diffcov_vs_bs2-1.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bs2-1.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.vs_bs2-1.rmet.log 2>&1 &


python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file shuidao2-1.guppy.pass2.CG.bn13_sn16.arab.balance.both_bilstm.10x_3.p0.8.freq.tsv --nano_file shuidao2-1.guppy.pass2.CG.bn13_sn16.rice2-1.balance.both_bilstm.10x_3.p0.8.freq.tsv --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.dp2_arab.n.dp2_rice2-1.10x_3.p0.8.corr_vs_bs.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file shuidao2-1.guppy.pass2.CHG.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --nano_file shuidao2-1.guppy.pass2.CHG.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHG.dp2_arab.n.dp2_rice2-1.10x_3.p0.8.corr_vs_bs.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file shuidao2-1.guppy.pass2.CHH.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --nano_file shuidao2-1.guppy.pass2.CHH.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHH.dp2_arab.n.dp2_rice2-1.10x_3.p0.8.corr_vs_bs.log &



# megalodon run rice2-1 ==============
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.rice2-1.pass2_guppy.10x_1.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_2.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_3.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_4.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.megalodon_bscov15.diffcov_vs_bs2-1.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.megalodon_bscov15.diffcov_vs_bs2-1.log 2>&1 &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.rice2-1.pass2_guppy.10x_1.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_2.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_3.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_4.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHG.megalodon_bscov15.diffcov_vs_bs2-1.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHG.megalodon_bscov15.diffcov_vs_bs2-1.log 2>&1 &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile megalodon_results.rice2-1.pass2_guppy.10x_1.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_2.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_3.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_4.CHH_retrain_comb/modified_bases.5mC.bed --nanofile megalodon_results.rice2-1.pass2_guppy.10x_5.CHH_retrain_comb/modified_bases.5mC.bed --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --repeat 5 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHH.megalodon_bscov15.diffcov_vs_bs2-1.tsv --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHH.megalodon_bscov15.diffcov_vs_bs2-1.log 2>&1 &

python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.rice2-1.pass2_guppy.10x_1.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_2.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_3.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_4.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile megalodon_results.rice2-1.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.rice2-1.pass2_guppy.10x_1.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_2.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_3.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_4.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile megalodon_results.rice2-1.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile megalodon_results.rice2-1.pass2_guppy.10x_1.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_2.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_3.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_4.CHH_retrain_comb/modified_bases.5mC.bed --bedfile megalodon_results.rice2-1.pass2_guppy.10x_5.CHH_retrain_comb/modified_bases.5mC.bed --wfile megalodon_results.rice2-1.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed &

python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.rice2-1.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice2-1.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.vs_bs2-1.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice2-1.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.vs_bs2-1.rmet.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.rice2-1.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice2-1.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.vs_bs2-1.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice2-1.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.vs_bs2-1.rmet.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/stats_correlation_with_bisulfite/performance_corr_heatmap.data.py --modsfile megalodon_results.rice2-1.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed --bsfile ../bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --wfile /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice2-1.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.vs_bs2-1.rmet.tsv > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/megalodon_results.rice2-1.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.vs_bs2-1.rmet.log 2>&1 &


# other tools, shuidao2-1 10x =
# deepsignal2 rerun m_arab/m_rice on shuidao2-1 10x.3
# m_arab
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.arab/model.dp2.CG.arab_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao2-1.guppy.pass2.CG.bn13_sn16.arab.balance.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass2.CG.bn13_sn16.arab.balance.both_bilstm.10x_3.log 2>&1
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.arab/model.dp2.CHG.arab_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch5.ckpt --result_file shuidao2-1.guppy.pass2.CHG.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao1-1.guppy.pass2.CHG.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.log 2>&1
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.arab/model.dp2.CHH.arab_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch7.ckpt --result_file shuidao2-1.guppy.pass2.CHH.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass2.CHH.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.log 2>&1

python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass2.CG.bn13_sn16.arab.balance.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass2.CG.bn13_sn16.arab.balance.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass2.CHG.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass2.CHG.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass2.CHH.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass2.CHH.bn13_sn16.arab.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &


# m_rice
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.dp2.CG.rice2-1_R9.4plus_tem.bn13_sn16.balance.both_bilstm/both_bilstm.b13_s16_epoch5.ckpt --result_file shuidao2-1.guppy.pass2.CG.bn13_sn16.rice2-1.balance.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CG --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass2.CG.bn13_sn16.rice2-1.balance.both_bilstm.10x_3.log 2>&1
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.dp2.CHG.rice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao2-1.guppy.pass2.CHG.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHG --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass2.CHG.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.log 2>&1
CUDA_VISIBLE_DEVICES=0 python ~/tools/deepsignal2/deepsignal2/call_modifications.py --input_path fast5_pass2.single/10x.3 --model_path model.dp2.CHH.rice2-1_R9.4plus_tem.bn13_sn16.denoise_signal_bilstm.both_bilstm/both_bilstm.b13_s16_epoch4.ckpt --result_file shuidao2-1.guppy.pass2.CHH.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --model_type both_bilstm --seq_len 13 --signal_len 16 --layernum1 3 --layernum2 1 --corrected_group RawGenomeCorrected_001 --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --motifs CHH --mod_loc 0 --f5_batch_size 20 --nproc 35 --nproc_gpu 6 > shuidao2-1.guppy.pass2.CHH.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.log 2>&1

python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass2.CG.bn13_sn16.rice2-1.balance.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass2.CG.bn13_sn16.rice2-1.balance.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass2.CHG.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass2.CHG.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass2.CHH.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.tsv --result_file shuidao2-1.guppy.pass2.CHH.bn13_sn16.rice2-1.denoise_signal_bilstm.both_bilstm.10x_3.p0.8.freq.tsv --prob_cf 0.8 &


# deepsignal CG
CUDA_VISIBLE_DEVICES=0 deepsignal call_mods --input_path fast5_pass2.single/10x.3 --model_path model.CpG.R9.4_1D.human_hx1.bn17.sn360.v0.1.7+/bn_17.sn_360.epoch_9.ckpt --reference_path Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --corrected_group RawGenomeCorrected_001 --nproc 30 --is_gpu yes --result_file shuidao2-1.guppy.pass2.CG.deepsignal.10x_3.tsv > shuidao2-1.guppy.pass2.CG.deepsignal.10x_3.log 2>&1 &
python /homeb/nipeng/tools/deepsignal2/scripts/call_modification_frequency.py --input_path shuidao2-1.guppy.pass2.CG.deepsignal.10x_3.tsv --result_file shuidao2-1.guppy.pass2.CG.deepsignal.10x_3.freq.tsv --prob_cf 0 &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_cpg_two_strands_methy_freqs.py --report_fp shuidao2-1.guppy.pass2.CG.deepsignal.10x_3.freq.tsv -t deepsignal --ref_fp /homeb/nipeng/data/genome/rice/Oryza_sativa.IRGSP-1.0.dna.toplevel.fa &



# tombo
tombo resquiggle fast5_pass2.single/10x.3 Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --processes 30 --corrected-group RawGenomeCorrected_001 --basecall-group Basecall_1D_001 --overwrite --ignore-read-locks > tombo_resquiggle.rice2-1.pass2_guppy.10x.3.log 2>&1 &
tombo detect_modifications alternative_model --fast5-basedirs fast5_pass2.single/10x.3 --alternate-bases CpG --statistics-file-basename tombo_results.rice2-1.pass2_guppy.10x.3.CG --dna --multiprocess-region-size 1000 --processes 30 --corrected-group RawGenomeCorrected_001 > tombo_results.rice2-1.pass2_guppy.10x.3.CG.log 2>&1 &
tombo text_output browser_files \
   --fast5-basedirs fast5_pass2.single/10x.3 \
   --statistics-filename tombo_results.rice2-1.pass2_guppy.10x.3.CG.CpG.tombo.stats \
   --file-types coverage dampened_fraction fraction\
   --browser-file-basename tombo_results.rice2-1.pass2_guppy.10x.3.CG --corrected-group RawGenomeCorrected_001 \
   > tombo_results.rice2-1.pass2_guppy.10x.3.CG.output.log 2>&1 &
wig2bed --multisplit < tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.minus.wig > tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.minus.wig.bed
wig2bed --multisplit < tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.plus.wig > tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.plus.wig.bed
python ~/tools/plant_5mC_analysis/tools_to_cmp/tombo_reformat_bed.py --cov_plus tombo_results.rice2-1.pass2_guppy.10x.3.CG.coverage.plus.bedgraph --cov_minus tombo_results.rice2-1.pass2_guppy.10x.3.CG.coverage.minus.bedgraph --rmet_plus tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.plus.wig.bed --rmet_minus tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.minus.wig.bed --wfile tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.methyl.bed &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_cpg_two_strands_methy_freqs.py --report_fp tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.methyl.bed -t bedmethyl --ref_fp /homeb/nipeng/data/genome/rice/Oryza_sativa.IRGSP-1.0.dna.toplevel.fa &
# =non-CG
tombo detect_modifications alternative_model --fast5-basedirs fast5_pass2.single/10x.3 --alternate-bases 5mC --statistics-file-basename tombo_results.rice2-1.pass2_guppy.10x.3.5mC --dna --multiprocess-region-size 1000 --processes 30 --corrected-group RawGenomeCorrected_001 > tombo_results.rice2-1.pass2_guppy.10x.3.5mC.log 2>&1 &
tombo text_output browser_files \
   --fast5-basedirs fast5_pass2.single/10x.3 \
   --statistics-filename tombo_results.rice2-1.pass2_guppy.10x.3.5mC.5mC.tombo.stats \
   --file-types coverage dampened_fraction fraction\
   --browser-file-basename tombo_results.rice2-1.pass2_guppy.10x.3.5mC --corrected-group RawGenomeCorrected_001 \
   > tombo_results.rice2-1.pass2_guppy.10x.3.5mC.output.log 2>&1 &
wig2bed --multisplit < tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.minus.wig > tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.minus.wig.bed &
wig2bed --multisplit < tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.plus.wig > tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.plus.wig.bed &
python ~/tools/plant_5mC_analysis/tools_to_cmp/tombo_reformat_bed.py --cov_plus tombo_results.rice2-1.pass2_guppy.10x.3.5mC.coverage.plus.bedgraph --cov_minus tombo_results.rice2-1.pass2_guppy.10x.3.5mC.coverage.minus.bedgraph --rmet_plus tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.plus.wig.bed --rmet_minus tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.minus.wig.bed --wfile tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.methyl.bed &


# nanopolish CG
guppy_basecaller -i fast5_pass2.single/10x.3 -r -s fast5_pass2.single.10x.3.guppy_fq --config dna_r9.4.1_450bps_hac_prom.cfg --gpu_runners_per_device 2 --chunks_per_runner 2500 --device CUDA:2 > guppy_fast5_pass2.single.10x.3.log &
/usr/bin/time -v bash nanopolish_call_methy.sh fast5_pass2.single/10x.3 fast5_pass2.single.10x.3.guppy_fq Oryza_sativa.IRGSP-1.0.dna.toplevel.fa 30 shuidao2-1.guppy.pass2.CG.nanopolish.10x_3 > shuidao2-1.guppy.pass2.CG.nanopolish.10x_3.log 2>&1 &
python /homeb/nipeng/tools/nanopolish/scripts/calculate_methylation_frequency.py -s fast5_pass2.single.10x.3.fastq.methyl_calls.tsv > shuidao2-1.guppy.pass2.CG.nanopolish.10x_3.methyl_calls.freq.tsv
python /homeb/nipeng/tools/nanopolish/scripts/calculate_methylation_frequency.s.py -s fast5_pass2.single.10x.3.fastq.methyl_calls.tsv > shuidao2-1.guppy.pass2.CG.nanopolish.10x_3.methyl_calls.freq.strand_specific.tsv

# megalodon 2.2.3 CG/CHG/CHH
megalodon fast5_pass2.single/10x.3/ \
    --guppy-server-path /usr/bin/guppy_basecall_server \
    --guppy-params "-d /home/nipeng/tools/rerio/basecall_models/ --num_callers 2" \
    --guppy-timeout 60 \
    --guppy-config res_dna_r941_prom_modbases_5mC_CpG_v001.cfg \
    --outputs per_read_mods mod_mappings mods \
    --reference Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --mod-motif m CG 0 \
    --write-mods-text \
    --mod-output-formats bedmethyl \
    --devices 0 --processes 20 --output-directory megalodon_results.rice2-1.pass.part2_guppy.10x_3.CG \
    --overwrite > megalodon_results.rice2-1.pass.part2_guppy.10x_3.CG.log 2>&1 &
megalodon fast5_pass2.single/10x.3/ \
    --guppy-server-path /opt/ont/guppy/bin/guppy_basecall_server \
    --guppy-params "-d /home/nipeng/tools/rerio/basecall_models/ --gpu_runners_per_device 2 --num_callers 6" \
    --guppy-timeout 60 \
    --guppy-config res_dna_r941_min_modbases-all-context_v001.cfg \
    --outputs per_read_mods mod_mappings mods \
    --reference Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --mod-motif Z CHG 0 \
    --write-mods-text \
    --mod-output-formats bedmethyl \
    --devices 1 --processes 20 --output-directory megalodon_results.rice2-1.pass.part2_guppy.10x_3.CHG \
    --overwrite > megalodon_results.rice2-1.pass.part2_guppy.10x_3.CHG.log 2>&1 &
megalodon fast5_pass2.single/10x.3/ \
    --guppy-server-path /opt/ont/guppy/bin/guppy_basecall_server \
    --guppy-params "-d /home/nipeng/tools/rerio/basecall_models/ --gpu_runners_per_device 2 --num_callers 6" \
    --guppy-timeout 60 \
    --guppy-config res_dna_r941_min_modbases-all-context_v001.cfg \
    --outputs per_read_mods mod_mappings mods \
    --reference Oryza_sativa.IRGSP-1.0.dna.toplevel.fa --mod-motif Z CHH 0 \
    --write-mods-text \
    --mod-output-formats bedmethyl \
    --devices 2 --processes 30 --output-directory megalodon_results.rice2-1.pass.part2_guppy.10x_3.CHH \
    --overwrite > megalodon_results.rice2-1.pass.part2_guppy.10x_3.CHH.log 2>&1 &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_cpg_two_strands_methy_freqs.py --report_fp megalodon_results.rice2-1.pass.part2_guppy.10x_3.CG/modified_bases.5mC.bed -t bedmethyl --ref_fp /homeb/nipeng/data/genome/rice/Oryza_sativa.IRGSP-1.0.dna.toplevel.fa &

python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_cpg_two_strands_methy_freqs.py --report_fp /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --rtype bs --ref_fp /homeb/nipeng/data/genome/rice/Oryza_sativa.IRGSP-1.0.dna.toplevel.fa &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_cpg_two_strands_methy_freqs.py --report_fp ../dp2_comb.results/shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.p0.8.freq.tsv --rtype deepsignal --ref_fp /homeb/nipeng/data/genome/rice/Oryza_sativa.IRGSP-1.0.dna.toplevel.fa &

python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.methyl.bed --nano_file shuidao2-1.guppy.pass2.CG.deepsignal.10x_3.freq.tsv --nano_file megalodon_results.rice2-1.pass.part2_guppy.10x_3.CG/modified_bases.5mC.bed --nano_file shuidao2-1.guppy.pass2.CG.nanopolish.10x_3.methyl_calls.freq.strand_specific.tsv --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.dp2_vs_othertools.10x_3.corr_vs_bs.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file shuidao2-1.guppy.pass2.CG.nanopolish.10x_3.methyl_calls.freq.strand_specific.tsv --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.dp2_vs_othertools.10x_3.corr_vs_bs.nanopolish_patch.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file tombo_results.rice2-1.pass2_guppy.10x.3.CG.dampened_fraction_modified_reads.methyl.fb_combined.bed --nano_file shuidao2-1.guppy.pass2.CG.deepsignal.10x_3.freq.fb_combined.tsv --nano_file megalodon_results.rice2-1.pass.part2_guppy.10x_3.CG/modified_bases.5mC.fb_combined.bed --nano_file shuidao2-1.guppy.pass2.CG.nanopolish.10x_3.methyl_calls.freq.tsv --nano_file ../dp2_comb.results/shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_3.p0.8.freq.fb_combined.tsv --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.fb_combined.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.dp2_vs_othertools.10x_3.corr_vs_bs.fb_comb.log &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file shuidao2-1.guppy.pass2.CG.nanopolish.10x_3.methyl_calls.freq.tsv --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.fb_combined.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CG.dp2_vs_othertools.10x_3.corr_vs_bs.fb_comb.nanopolish_patch.log &

python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.methyl.bed --nano_file megalodon_results.rice2-1.pass.part2_guppy.10x_3.CHG/modified_bases.5mC.bed --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHG.dp2_vs_othertools.10x_3.corr_vs_bs.log &

python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file tombo_results.rice2-1.pass2_guppy.10x.3.5mC.dampened_fraction_modified_reads.methyl.bed --nano_file megalodon_results.rice2-1.pass.part2_guppy.10x_3.CHH/modified_bases.5mC.bed --bs_file /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/shuidao2-1.guppy.pass2.CHH.dp2_vs_othertools.10x_3.corr_vs_bs.log &




# other tools, arab 10x CG fb_comb patch =
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_cpg_two_strands_methy_freqs.py --report_fp athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.p0.8.freq.tsv --rtype deepsignal --ref_fp /homeb/nipeng/data/genome/athaliana/GCF_000001735.4_TAIR10.1_genomic.fna &

python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/correlation_with_bs.py --nano_file athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.10x_1.p0.8.freq.fb_combined.tsv --bs_file ../bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.fb_combined.txt --bs_file ../bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.fb_combined.txt --bs_file ../bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.fb_combined.txt --contig_prefix NC_003 --cov_cf 5 > /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/athaliana.guppy.pass.part2.CG.10x_1.cmp_tools.vs_bs_rep123.stat.pearson_corr.fb_comb.dp2_p0.8.log &






# megalodon bs_cov_15 bnigra ===
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CG/ni-100_BP_1D_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CG/ni-100_BP_30kb_1D_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CG/ni-100_BP30kb_2_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile ni-100_10x1_CG.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CG/ni-100_BP_1_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CG/ni-100_BP_2_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CG/ni-100Gtube_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile ni-100_10x2_CG.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CG/ni-100_BP_Rapid_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CG/ni-100_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile ni-100_10x3_CG.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CG/ni-100Qiagen_fast5.CG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile ni-100_10xelse_CG.megalodon_bscov15.modified_bases.5mC.bed &

python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CHG/ni-100_BP_1D_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CHG/ni-100_BP_30kb_1D_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CHG/ni-100_BP30kb_2_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile ni-100_10x1_CHG.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CHG/ni-100_BP_1_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CHG/ni-100_BP_2_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CHG/ni-100Gtube_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile ni-100_10x2_CHG.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CHG/ni-100_BP_Rapid_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --bedfile CHG/ni-100_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile ni-100_10x3_CHG.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CHG/ni-100Qiagen_fast5.CHG_retrain_comb.bs_cov_15/modified_bases.5mC.bed --wfile ni-100_10xelse_CHG.megalodon_bscov15.modified_bases.5mC.bed &

python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CHH/ni-100_BP_1D_fast5.CHH_retrain_comb/modified_bases.5mC.bed --bedfile CHH/ni-100_BP_30kb_1D_fast5.CHH_retrain_comb/modified_bases.5mC.bed --bedfile CHH/ni-100_BP30kb_2_fast5.CHH_retrain_comb/modified_bases.5mC.bed --wfile ni-100_10x1_CHH.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CHH/ni-100_BP_1_fast5.CHH_retrain_comb/modified_bases.5mC.bed --bedfile CHH/ni-100_BP_2_fast5.CHH_retrain_comb/modified_bases.5mC.bed --bedfile CHH/ni-100Gtube_fast5.CHH_retrain_comb/modified_bases.5mC.bed --wfile ni-100_10x2_CHH.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CHH/ni-100_BP_Rapid_fast5.CHH_retrain_comb/modified_bases.5mC.bed --bedfile CHH/ni-100_fast5.CHH_retrain_comb/modified_bases.5mC.bed --wfile ni-100_10x3_CHH.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile CHH/ni-100Qiagen_fast5.CHH_retrain_comb/modified_bases.5mC.bed --wfile ni-100_10xelse_CHH.megalodon_bscov15.modified_bases.5mC.bed &

python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile ni-100_10x1_CG.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10x2_CG.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10x3_CG.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10xelse_CG.megalodon_bscov15.modified_bases.5mC.bed --wfile ni-100_total_CG.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile ni-100_10x1_CHG.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10x2_CHG.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10x3_CHG.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10xelse_CHG.megalodon_bscov15.modified_bases.5mC.bed --wfile ni-100_total_CHG.megalodon_bscov15.modified_bases.5mC.bed &
python3 /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/combine_methylbed_files.py --bedfile ni-100_10x1_CHH.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10x2_CHH.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10x3_CHH.megalodon_bscov15.modified_bases.5mC.bed --bedfile ni-100_10xelse_CHH.megalodon_bscov15.modified_bases.5mC.bed --wfile ni-100_total_CHH.megalodon_bscov15.modified_bases.5mC.bed &

python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile ni-100_10x1_CG.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10x2_CG.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10x3_CG.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10xelse_CG.megalodon_bscov15.modified_bases.5mC.bed --bsfile ../bs.poses/Ni100-ox1-Bnigra.v3.NI100.methratio.methylbed.CG.bed --repeat 6 --wfile /homeb/nipeng/tools/plant_5mC_analysis/public_data_test/bnigra_ni100.correlation_vs_bs.CG.diffcov.megalodon_bscov15.txt --contig_names B1,B2,B3,B4,B5,B6,B7,B8 --cov_cf 5 --cov_cf_bs 10 &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile ni-100_10x1_CHG.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10x2_CHG.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10x3_CHG.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10xelse_CHG.megalodon_bscov15.modified_bases.5mC.bed --bsfile ../bs.poses/Ni100-ox1-Bnigra.v3.NI100.methratio.methylbed.CHG.bed --repeat 6 --wfile /homeb/nipeng/tools/plant_5mC_analysis/public_data_test/bnigra_ni100.correlation_vs_bs.CHG.diffcov.megalodon_bscov15.txt --contig_names B1,B2,B3,B4,B5,B6,B7,B8 --cov_cf 5 --cov_cf_bs 10 &
python /homeb/nipeng/tools/plant_5mC_analysis/tools_to_cmp/performance_coverage_effect.py --nanofile ni-100_10x1_CHH.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10x2_CHH.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10x3_CHH.megalodon_bscov15.modified_bases.5mC.bed --nanofile ni-100_10xelse_CHH.megalodon_bscov15.modified_bases.5mC.bed --bsfile ../bs.poses/Ni100-ox1-Bnigra.v3.NI100.methratio.methylbed.CHH.bed --repeat 6 --wfile /homeb/nipeng/tools/plant_5mC_analysis/public_data_test/bnigra_ni100.correlation_vs_bs.CHH.diffcov.megalodon_bscov15.txt --contig_names B1,B2,B3,B4,B5,B6,B7,B8 --cov_cf 5 --cov_cf_bs 10 &






# analysis ===========
# arab bio1, 254.2
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile ~/tools/data/arab/bs.result/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ~/tools/data/arab/bs.result/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile ~/tools/data/arab/bs.result/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --wfile athaliana.methyfreq_bins.CG.bsrep123_inter_cov5_main_contigs.tsv --covcf 5 --contig_prefix NC_003 > athaliana.methyfreq_bins.CG.bsrep123_inter_cov5_main_contigs.log &
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile ~/tools/data/arab/bs.result/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ~/tools/data/arab/bs.result/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile ~/tools/data/arab/bs.result/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --wfile athaliana.methyfreq_bins.CHG.bsrep123_inter_cov5_main_contigs.tsv --covcf 5 --contig_prefix NC_003 > athaliana.methyfreq_bins.CHG.bsrep123_inter_cov5_main_contigs.log &
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile ~/tools/data/arab/bs.result/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ~/tools/data/arab/bs.result/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile ~/tools/data/arab/bs.result/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --wfile athaliana.methyfreq_bins.CHH.bsrep123_inter_cov5_main_contigs.tsv --covcf 5 --contig_prefix NC_003 > athaliana.methyfreq_bins.CHH.bsrep123_inter_cov5_main_contigs.log &

python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin athaliana.methyfreq_bins.CG.bsrep123_inter_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/deepsignal2_results/athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile athaliana.methyfreq_bins.CG.deepsignal_comb_p0.8_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin athaliana.methyfreq_bins.CHG.bsrep123_inter_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/deepsignal2_results/athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile athaliana.methyfreq_bins.CHG.deepsignal_comb_p0.8_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin athaliana.methyfreq_bins.CHH.bsrep123_inter_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/deepsignal2_results/athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile athaliana.methyfreq_bins.CHH.deepsignal_comb_p0.8_cov5_main_contigs.tsv &

python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin athaliana.methyfreq_bins.CG.bsrep123_inter_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/megalodon_results/megalodon_results.arab.pass.part2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile athaliana.methyfreq_bins.CG.megalodon_bs_cov_15_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin athaliana.methyfreq_bins.CHG.bsrep123_inter_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/megalodon_results/megalodon_results.arab.pass.part2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile athaliana.methyfreq_bins.CHG.megalodon_bs_cov_15_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin athaliana.methyfreq_bins.CHH.bsrep123_inter_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/megalodon_results/megalodon_results.arab.pass.part2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile athaliana.methyfreq_bins.CHH.megalodon_bs_cov_15_cov5_main_contigs.tsv &

cat athaliana.methyfreq_bins.CG.[bdm]*tsv > athaliana.methyfreq_bins.CG.comb.dp2p0.8_megbscov15.tsv &
cat athaliana.methyfreq_bins.CHG.[bdm]*tsv > athaliana.methyfreq_bins.CHG.comb.dp2p0.8_megbscov15.tsv &
cat athaliana.methyfreq_bins.CHH.[bdm]*tsv > athaliana.methyfreq_bins.CHH.comb.dp2p0.8_megbscov15.tsv &


python comparison_highly_lowly_methylated_sites.py --contig_prefix NC_003 --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --freqfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/deepsignal2_results/athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --wfile athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > athaliana.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &
python comparison_highly_lowly_methylated_sites.py --contig_prefix NC_003 --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --freqfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/deepsignal2_results/athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --wfile athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > athaliana.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &
python comparison_highly_lowly_methylated_sites.py --contig_prefix NC_003 --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902826A-ZJ_301301.L1L3_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902827A-ZJ_364364.L1L2_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --bsfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/bs.poses/D1902828A-ZJ_363363.L2L4_merged.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --freqfile /homeb/nipeng/data/nanopore/arabidopsis/20190421-NPL0867-P1-E11-H11-barcode/deepsignal2_results/athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --wfile athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > athaliana.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &



# rice2-1 254.3
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --wfile rice2-1.methyfreq_bins.CG.bs2-1_cov5_main_contigs.tsv --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > rice2-1.methyfreq_bins.CG.bs2-1_cov5_main_contigs.log &
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --wfile rice2-1.methyfreq_bins.CHG.bs2-1_cov5_main_contigs.tsv --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > rice2-1.methyfreq_bins.CHG.bs2-1_cov5_main_contigs.log &
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --wfile rice2-1.methyfreq_bins.CHH.bs2-1_cov5_main_contigs.tsv --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > rice2-1.methyfreq_bins.CHH.bs2-1_cov5_main_contigs.log &

python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice2-1.methyfreq_bins.CG.bs2-1_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/dp2_comb.results/shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile rice2-1.methyfreq_bins.CG.deepsignal2_comb_p0.8_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice2-1.methyfreq_bins.CHG.bs2-1_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/dp2_comb.results/shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile rice2-1.methyfreq_bins.CHG.deepsignal2_comb_p0.8_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice2-1.methyfreq_bins.CHH.bs2-1_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/dp2_comb.results/shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile rice2-1.methyfreq_bins.CHH.deepsignal2_comb_p0.8_cov5_main_contigs.tsv &

python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice2-1.methyfreq_bins.CG.bs2-1_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/megalodon.results/megalodon_results.rice2-1.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile rice2-1.methyfreq_bins.CG.megalodon_bs_cov_15_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice2-1.methyfreq_bins.CHG.bs2-1_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/megalodon.results/megalodon_results.rice2-1.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile rice2-1.methyfreq_bins.CHG.megalodon_bs_cov_15_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice2-1.methyfreq_bins.CHH.bs2-1_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/megalodon.results/megalodon_results.rice2-1.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile rice2-1.methyfreq_bins.CHH.megalodon_bs_cov_15_cov5_main_contigs.tsv &

cat rice2-1.methyfreq_bins.CG.[bdm]*tsv > rice2-1.methyfreq_bins.CG.comb.dp2p0.8_megbscov15.tsv &
cat rice2-1.methyfreq_bins.CHG.[bdm]*tsv > rice2-1.methyfreq_bins.CHG.comb.dp2p0.8_megbscov15.tsv &
cat rice2-1.methyfreq_bins.CHH.[bdm]*tsv > rice2-1.methyfreq_bins.CHH.comb.dp2p0.8_megbscov15.tsv &


python comparison_highly_lowly_methylated_sites.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --freqfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/dp2_comb.results/shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --wfile shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > shuidao2-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &
python comparison_highly_lowly_methylated_sites.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --freqfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/dp2_comb.results/shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --wfile shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > shuidao2-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &
python comparison_highly_lowly_methylated_sites.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/bs.poses/shuidao2-1_1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --freqfile /homeb/nipeng/data/nanopore/rice/shuidao2-1/fast5_pass/dp2_comb.results/shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --wfile shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > shuidao2-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &




# rice1-1 bio1, 254.3
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile ~/tools/data/rice/bs.result/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --wfile rice1-1.methyfreq_bins.CG.bs1-2_cov5_main_contigs.tsv --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > rice1-1.methyfreq_bins.CG.bs1-2_cov5_main_contigs.log &
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile ~/tools/data/rice/bs.result/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --wfile rice1-1.methyfreq_bins.CHG.bs1-2_cov5_main_contigs.tsv --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > rice1-1.methyfreq_bins.CHG.bs1-2_cov5_main_contigs.log &
python3 comparison_highly_lowly_methylated_sites2_bs.py --bsfile ~/tools/data/rice/bs.result/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --wfile rice1-1.methyfreq_bins.CHH.bs1-2_cov5_main_contigs.tsv --covcf 5 --contig_names 1,2,3,4,5,6,7,8,9,10,11,12 > rice1-1.methyfreq_bins.CHH.bs1-2_cov5_main_contigs.log &

python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice1-1.methyfreq_bins.CG.bs1-2_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/deepsignal2_comb.result/shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile rice1-1.methyfreq_bins.CG.deepsignal2_comb_p0.8_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice1-1.methyfreq_bins.CHG.bs1-2_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/deepsignal2_comb.result/shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile rice1-1.methyfreq_bins.CHG.deepsignal2_comb_p0.8_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice1-1.methyfreq_bins.CHH.bs1-2_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/deepsignal2_comb.result/shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --cov_cf 5 --techname deepsignal2 --wfile rice1-1.methyfreq_bins.CHH.deepsignal2_comb_p0.8_cov5_main_contigs.tsv &

python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice1-1.methyfreq_bins.CG.bs1-2_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/megalodon.result/megalodon_results.rice.pass2_guppy.50x_12345.CG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile rice1-1.methyfreq_bins.CG.megalodon_bs_cov_15_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice1-1.methyfreq_bins.CHG.bs1-2_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/megalodon.result/megalodon_results.rice.pass2_guppy.50x_12345.CHG_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile rice1-1.methyfreq_bins.CHG.megalodon_bs_cov_15_cov5_main_contigs.tsv &
python3 comparison_highly_lowly_methylated_sites2_nano.py --bsbin rice1-1.methyfreq_bins.CHH.bs1-2_cov5_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/megalodon.result/megalodon_results.rice.pass2_guppy.50x_12345.CHH_retrain_comb.bs_cov_15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile rice1-1.methyfreq_bins.CHH.megalodon_bs_cov_15_cov5_main_contigs.tsv &

cat rice1-1.methyfreq_bins.CG.[bdm]*tsv > rice1-1.methyfreq_bins.CG.comb.dp2p0.8_megbscov15.tsv &
cat rice1-1.methyfreq_bins.CHG.[bdm]*tsv > rice1-1.methyfreq_bins.CHG.comb.dp2p0.8_megbscov15.tsv &
cat rice1-1.methyfreq_bins.CHH.[bdm]*tsv > rice1-1.methyfreq_bins.CHH.comb.dp2p0.8_megbscov15.tsv &


python comparison_highly_lowly_methylated_sites.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CG.txt --freqfile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/deepsignal2_comb.result/shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.tsv --wfile shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > shuidao1-1.guppy.pass.part2.CG.bn13_sn16.arabnrice2-1.balance.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &
python comparison_highly_lowly_methylated_sites.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHG.txt --freqfile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/deepsignal2_comb.result/shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --wfile shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > shuidao1-1.guppy.pass.part2.CHG.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &
python comparison_highly_lowly_methylated_sites.py --bsfile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/bs.poses/D1904161A-QJ_L2L4.cutadapt.R1_bismark_bt2_pe.sorted.mark_dup.sorted.CX_report.CHH.txt --freqfile /homeb/nipeng/data/nanopore/rice/shuidao1-1/fast5/deepsignal2_comb.result/shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.tsv --wfile shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.tsv > shuidao1-1.guppy.pass.part2.CHH.bn13_sn16.arabnrice2-1.denoise_signal_bilstm.both_bilstm.50x_12345.p0.8.freq.high_low_in_bs.log &



# bnigra 254.6
python3 ../tools_to_cmp/comparison_highly_lowly_methylated_sites2_nano.py --bsbin bnigra_ni100.methyfreq_bins.CG.bsmap_cov10_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/bnigra/dp_results/megalodon_bscov15/ni-100_total_CG.megalodon_bscov15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile bnigra_ni100.methyfreq_bins.CG.megalodonbscov15_cov5_vsbsmap_cov10_main_contigs.tsv &
python3 ../tools_to_cmp/comparison_highly_lowly_methylated_sites2_nano.py --bsbin bnigra_ni100.methyfreq_bins.CHG.bsmap_cov10_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/bnigra/dp_results/megalodon_bscov15/ni-100_total_CHG.megalodon_bscov15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile bnigra_ni100.methyfreq_bins.CHG.megalodonbscov15_cov5_vsbsmap_cov10_main_contigs.tsv &
python3 ../tools_to_cmp/comparison_highly_lowly_methylated_sites2_nano.py --bsbin bnigra_ni100.methyfreq_bins.CHH.bsmap_cov10_main_contigs.tsv --nanofile /homeb/nipeng/data/nanopore/bnigra/dp_results/megalodon_bscov15/ni-100_total_CHH.megalodon_bscov15.modified_bases.5mC.bed --cov_cf 5 --techname megalodon --wfile bnigra_ni100.methyfreq_bins.CHH.megalodonbscov15_cov5_vsbsmap_cov10_main_contigs.tsv &

cat bnigra_ni100.methyfreq_bins.CG.[bdm]*tsv > bnigra_ni100.methyfreq_bins.CG.comb.dp2p0.8_megbscov15.tsv &
cat bnigra_ni100.methyfreq_bins.CHG.[bdm]*tsv > bnigra_ni100.methyfreq_bins.CHG.comb.dp2p0.8_megbscov15.tsv &
cat bnigra_ni100.methyfreq_bins.CHH.[bdm]*tsv > bnigra_ni100.methyfreq_bins.CHH.comb.dp2p0.8_megbscov15.tsv &
















